src/main/java/uk/gov/pay/api/model/SelfLinks.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.pay.api.model;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+import static javax.ws.rs.HttpMethod.GET;
+
+@ApiModel(value = "selfLinks", description = "Resource self link of a Payment")
+public class SelfLinks {
```

Comment:
I thought the class described what its suppose to do if we name it as 'SelfLinks' and 'SelfAndNextLinks'. I've been able to use the same model when fixing up the PaymentEvents as well. I do agree _if_ we have to add a new link to the '_link' structure then we have to either add a new one or amend the name. But as of now I'm not aware of any reason to do this, other than at some point we'd want to remove '_self' from '_links' and make it its own object. In that event we can rename the 'SlefAndNextLinks' to 'Links', remove 'SelfLinks' and move the whole '_self' object into payments. But there's no assurances _if/when_ that would happen. So I thought of making the naming very specific - just following YAGNI.
If you still feel strongly about the naming I'm happy to change it as well though.

Another thing worth noting is that 'Link' class currently exists in this PR for the purpose of PaymentEvents. This is going away when PaymentEvents are fixed as it has its own 'PaymentEventLink' object. See https://payments-platform.atlassian.net/browse/PP-539 for details.